### DISCUSSION TITLE
grblHAL on Arduino Portenta H7

### DISCUSSION THREAD

**Discussion by the13thparish** [grblHAL/core#17] [Q&A]
Dear all,

Since there is a pre-existing grblHAL implementation for the STM32F4 ARM-M4 Cortex, I'm interested in running grblHAL on the M4 core of the Arduino Portenta H7, which has a STM32H747 dual-core MCU. I will leave the M7 core to do some sensor readings. 

1.) Do I need to download the core folder _and_ the STM32F4xx driver folder _and_ any plugins I might want? Or does the STM32F4xx folder contain everything I need if I only want to work on one particular board?

2.) For importation to the Arduino IDE as a library, where is the grblHAL_xxx folder mentioned [here](https://github.com/grblHAL/core/wiki/Compiling-GrblHAL)? I notice within the STM32F4xx folder and the core folder, there is a directory called grbl, but not grblHAL_xxx.

All the best,
Ryan

**Comment by terjeio** [comment]
> 1.) Do I need to download the core folder and the STM32F4xx driver folder and any plugins I might want? Or does the STM32F4xx folder contain everything I need if I only want to work on one particular board?

The STM32F4xx project contains all the folders needed, some as git submodules. But it is unlikely you can use the STM32F4xx driver "as-is", initialisation in `main()` is certainly going to be different. With luck the F4 and H7 peripherals are similar and if so the STM32F4xx driver code can serve as a template. Note that the driver is written on top ST libraries, Arduino Portenta H7 Arduino libraries on top of mbed - this could mean that you have to rewrite quite a bit of driver code.

> 2.) For importation to the Arduino IDE as a library, where is the grblHAL_xxx folder mentioned here? I notice within the STM32F4xx folder and the core folder, there is a directory called grbl, but not grblHAL_xxx.

There is none as this driver is built using the STM32CubeIDE which is Eclipse based.

**Reply by terjeio** [reply]
FYI there is now a [driver available for the ST32H7xx](https://github.com/grblHAL/STM32F7xx/discussions/5) family of chips - but it is not for the Arduino IDE.
